infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InitializedState.java

```diff
@@ -77,8 +77,13 @@ public void onEntry(LogReplicationState from) {
 
     @Override
     public void onExit(LogReplicationState to) {
+        LogReplicationStateType type = to.getType();
+        LogReplicationMetadata.SyncType toSyncType =
+                type.equals(LogReplicationStateType.IN_SNAPSHOT_SYNC) ? LogReplicationMetadata.SyncType.SNAPSHOT :
+                type.equals(LogReplicationStateType.IN_LOG_ENTRY_SYNC) ? LogReplicationMetadata.SyncType.LOG_ENTRY :
+                null;
```

Comment:
the "toSyncType" is for setting the value for `lastSyncType` in ack reader right? then if the to state is `WAIT_SNAPSHOT_APPLY` in case of SNAPSHOT_TRANSFER_COMPLETE, we will set it be null.
but shouldn't it be SyncType.SNAPSHOT I think?

Reply:
Yes you're right, we update the sync status as SyncType.SNAPSHOT for `WaitSnapshotApplyState` when coming from LogReplicationStateType.INITIALIZED in `markSnapshotSyncInfoOngoing()` so it makes sense setting it to SyncType.SNAPSHOT